crates/meilitool/src/main.rs

```diff
@@ -254,18 +252,39 @@ fn export_a_dump(
                 if status == Status::Enqueued {
                     let content_file = file_store.get_update(content_file_uuid)?;
 
-                    let reader =
-                        DocumentsBatchReader::from_reader(content_file).with_context(|| {
-                            format!("While reading content file {:?}", content_file_uuid)
-                        })?;
-
-                    let (mut cursor, documents_batch_index) = reader.into_cursor_and_fields_index();
-                    while let Some(doc) = cursor.next_document().with_context(|| {
-                        format!("While iterating on content file {:?}", content_file_uuid)
-                    })? {
-                        dump_content_file
-                            .push_document(&obkv_to_object(doc, &documents_batch_index)?)?;
+                    if (
+                        detected_version.0.as_str(),
+                        detected_version.1.as_str(),
+                        detected_version.2.as_str(),
+                    ) < ("1", "12", "0")
```

Comment:
Nice, I didn't know we could do that

Reply:
this is using the blanked Ord impl of `(T,U,V)` for `T, U, V: Ord`. https://doc.rust-lang.org/std/cmp/trait.Ord.html#impl-Ord-for-(T,)

It is doing lexicographic comparisons, which is what we want.

Reply:
:fire:

Reply:
That will work only before we release releases like 1.100.0 as _100 < 12_, for example.